certbot-ci/certbot_integration_tests/utils/acme_server.py

```diff
@@ -223,7 +232,7 @@ def _prepare_http_proxy(self):
         print('=> Configuring the HTTP proxy...')
         mapping = {r'.+\.{0}\.wtf'.format(node): 'http://127.0.0.1:{0}'.format(port)
                    for node, port in self.acme_xdist['http_port'].items()}
-        command = [sys.executable, proxy.__file__, str(HTTP_01_PORT), json.dumps(mapping)]
+        command = [sys.executable, proxy.__file__, str(DEFAULT_HTTP_01_PORT), json.dumps(mapping)]
```

Comment:
If you pass the user-defined http-01 port here, I think it will make it compatible with the proxy mode.